astropy/io/ascii/tests/test_read.py

```diff
@@ -1283,3 +1284,27 @@ def test_unsupported_read_with_encoding(tmpdir):
     with pytest.raises(ascii.ParameterError):
         ascii.read('t/simple3.txt', guess=False, fast_reader='force',
                    encoding='latin1', format='fast_csv')
+
+
+def test_read_pandas_csv(tmpdir):
+    pandas = pytest.importorskip('pandas')
+
+    df = pandas.DataFrame()
+    df['stuff'] = [x for x in range(10)]
+    no_index_fname = str(tmpdir.join('no_index.csv'))
+    df.to_csv(no_index_fname, index=False)
+
+    table = ascii.read(no_index_fname)
+    assert table.colnames == ['stuff']
+    assert all(table['stuff'] == [x for x in range(10)])
+
+    # This is the problematic case since by default the first column in a
+    # pandas csv is the index column. However, if we don't explicitly provide
+    # an index name, then the header column starts with a single delimiter,
+    # indicating an index header name of None
+    has_index_fname = str(tmpdir.join('has_index.csv'))
+    df.to_csv(has_index_fname)
+
+    table = ascii.read(has_index_fname)
```

Comment:
It's probably worth explicitly testing two different `format` values `basic` and `csv` in addition to the current test.  Your current test doesn't hit the `FastCsv` reader, only `FastBasic`.

One good practice when working on changes like this is always look at `ascii.get_read_trace()` after doing a read in order to see which readers were tried and (importantly) the one that was actually used for the final output.  The guess logic is complicated...

Reply:
@taldcroft thanks for the review. Clearly this PR needs some more work, so I'll let you know when it's ready for review again. The guess logic does seem complicated, and it's hard to know how many different places this kind of change needs to be implemented and tested.